### DISCUSSION TITLE
Offline access to data

### DISCUSSION THREAD

**Discussion by Kubel259** [theOehrly/Fast-F1#666] [General]
Hi, tomorrow I am flying back home and I won't be able to have internet access on the plane. However I'd like to play with the code and access the data. Let's say I run my code while still having internet connection at the airport, so that I download the data for a particular driver, year, race and session. It saves automatically in the Cache folder. Is is possible to have access to this data after I take off (no internet), so that I can still work on my project?

**Comment by theOehrly** [comment]
In case it is still helpful and I wasn't too slow to respond, yes that is possible. See https://docs.fastf1.dev/fastf1.html#fastf1.Cache.offline_mode

**Reply by Kubel259** [reply]
Great, thanks a lot. Right on time :)

**Reply by ruilov** [reply]
jolpica appears to be down right now, which alerted me to the fact that offline mode wasn't fully working for me because I was calling it before setting the cache directory.

```
import fastf1
fastf1.Cache.offline_mode(True)
print(fastf1.Cache._requests_session_cached.settings.only_if_cached)
fastf1.Cache.enable_cache("my_fastf1_cache")
print(fastf1.Cache._requests_session_cached.settings.only_if_cached)
```

only_if_cached gets set on the default cache. Then when I set the cache directory, that setting is never toggled to true. Perhaps this is the desired behavior but caught me out. And hopefully didn't result in me bringing jolpica down :(

**Reply by theOehrly** [reply]
@ruilov that is the expected behaviour right now. This should probably be noted in the documentation.

The Jolpica outage certainly wasn't caused by you but by an unexpected change by AWS which blocked all traffic.
